### ISSUE TITLE
[BUG] Modifiers get lost (kinda) when copying over local profile from SM 5.1

### ISSUE THREAD

**Issue by natano** [TeamRizu/OutFox#168]
**Please give System Info**

OS: Linux Mint 20

CPU: Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz

GPU: Intel Corporation HD Graphics 620 (rev 02)

Storage: Some SSD

**Please give your StepMania Info**

StepMania Version: 5.3 alpha 4.9.1
Theme: Simply Love ([wip-outfox branch](https://github.com/Simply-Love/Simply-Love-SM5/tree/wip-outfox))

**Describe the problem**

For reference: the profile's `Stats.xml` contains
```
<DefaultModifiers>
<dance>C675, Overhead, Cel</dance>
</DefaultModifiers>
```
and `Simply Love UserPrefs.ini` contains
```
PlayerOptionsString=C675, Overhead, Cel
SpeedMod=675
SpeedModType=C
```
This is where things get weird.

Reproduction:

1. Select the newly copied over profile after starting outfox.
2. Select a song and hit enter. **The arrows now scroll at 1x speed and the Wave acceleration is on.** I never enabled Wave anywhere and the speed should be C675.
3. After failing/passing the song, start it again, but now hit enter twice to enter the user options.
4. The options show that C675 is selected.
5. Go to the uncommon options. The Wave acceleration is on. Disable it.
6. Start the song. Now the speed is correct and wave acceleration is off. When you play more songs in the same session the options keep being correct.
7. Quit out of the game. Start it again and select the broken profile. You are now back at step 2.

This issue does not happen with all profiles. Apparently there has to be something in the profile to trigger this behaviour. I included a broken and a working profile with the issue, both created and played with 5.1.

**Describe what should happen**

The speed should be at C675 and Wave acceleration should be off.
The modifiers should be remembered across restarts of the game.

**Logs and Additional context**

Logs: [Logs.tar.gz](https://github.com/TeamRizu/OutFox/files/5550024/Logs.tar.gz)

A 5.1 profile where the issue happens: [00000000.tar.gz](https://github.com/TeamRizu/OutFox/files/5549982/00000000.tar.gz)

A 5.1 profile where the issue does **not** happen: [00000001.tar.gz](https://github.com/TeamRizu/OutFox/files/5549902/00000001.tar.gz)

**Comment by JoseVarelaP** [comment]
I've done some tests, and It may be due to the fact that the profile that causes the bug is loading an invalid noteskin case, as an example, I loaded the game without a Cel noteskin (as i don't even have one).

The following is a test of how data is being read in the xml, followed by what the game interprets.
```
Profile::LoadAllFromDir( /Save/LocalProfiles/00000006/ )
Loading /Save/LocalProfiles/00000006/Stats.xml
Done.
dance = C675, Overhead, Cel / [1x] (This is due to the Cel not making sense for the parser, so it returns the default modifiers set upon that point).
```

So i decided to use what i have on my own profile as settings with a CMod, `C600, Overhead, Outfox-note`, and loaded that.

```
Profile::LoadAllFromDir( /Save/LocalProfiles/00000006/ )
Loading /Save/LocalProfiles/00000006/Stats.xml
Done.
dance = C600, Overhead, Outfox-note / [C600, outfox-note]
```

With no issue, it was up an running. Still unsure where you're getting Wave from, as we have no refference of it being loaded as a Default Modifier. You should check your Preferences.ini for the DefaultModifiers (there are several per game) and your theme's DefaultModifiers (yep, they can set them too).

**Comment by natano** [comment]
FWIW, I have the Cel noteskin installed for outfox and it also shows up as an available option in game.
(Let me know if you need the noteskin files, I can send you a tarball.)

Here are all the preferences files:
[OutFoxPrefs.ini.txt](https://github.com/TeamRizu/OutFox/files/5563155/OutFoxPrefs.ini.txt)
[Preferences.ini.txt](https://github.com/TeamRizu/OutFox/files/5563157/Preferences.ini.txt)
[ThemePrefs.ini.txt](https://github.com/TeamRizu/OutFox/files/5563158/ThemePrefs.ini.txt)

Searching for the word "Wave" in the Save directory doesn't yield any relevant results either.
```
natano@lvalue:~/.stepmania-5.3/Save$ grep -Rw Wave .
./LocalProfiles/00000000/ECFA-Stats.xml:<Song Dir='Songs/The Starter Pack of Stamina/The Ninth Wave - [Braeden&amp;Tuuc]/'>
```

**Comment by natano** [comment]
I managed to reproduce the issue without Simply Love, but with the **Soundwaves** theme and a **new profile**:
* Create a new profile
* Start a game with that profile
* Set speed mod to C675 and play a chart (make sure your profile gets saved)
* Restart the game and start a game with that profile again
* Your speed mod is now 1x and the Wave acceleration is on

**C675 is special.** This doesn not happen with C650 or C700. :man_shrugging: 

The profile that I uploaded or any custom noteskins are not required to reproduce this issue.

**Comment by lomdr** [comment]
I tried to use C680 on a 'bugged profile' (A profile that used C675 previously) and it was able to save my commonly used mods (C680, Overhead) with a non-stock noteskin (starlight-smnote).  So it's specifically C675 being bugged, not the profile handlers itself.

**Comment by JoseVarelaP** [comment]
Looking more over it, my theory now seems it has to do in how the speedmod system was made before AMod was even a thing, given it's unsual implementation, and lack of updates over the years. Gonna look into that.

**Comment by Scraticus** [comment]
should be fixed in the next alpha - will leave open for the moment

**Comment by natano** [comment]
I can confirm this is fixed on 4.9.7 (pre hotfix).
